include/mbedtls/check_config.h

```diff
@@ -551,6 +551,16 @@
 #error "MBEDTLS_PK_WRITE_C defined, but not all prerequisites"
 #endif
 
+#if defined(MBEDTLS_PK_PARSE_C) && defined(MBEDTLS_PK_HAVE_ECC_KEYS) && \
```

Comment:
Is this dependency on OID really specific to ECC keys? I can see that `pk_get_pk_alg()` calls ` mbedtls_oid_get_pk_alg()` on a path that isn't guarded by `MBEDTLS_PK_HAVE_ECC_KEYS` for example.

Symmetrically, in pkwrite, we call `mbedtls_oid_get_oid_by_pk_alg()` on a path that doesn't seem specific to ECC.

Reply:
Ah OK, I had missed that, thanks for the detailed explanation!

I think it would still be better to depend on OID directly, since we use it directly. That way, if some day RSA stops depending on OID, then we still get proper dependencies for PK.

Reply:
My impression is that pkparse needs `mbedtls_oid_get_pk_alg()` for everything that matters and pkwrite needs `mbedtls_oid_get_oid_by_pk_alg()` for everything that matters. I agree that other uses of the OID module are guarded by other things (like PKCS5/12 support) but those two look pretty essential to me.

Let's wait for @gilles-peskine-arm 's input on this.

Reply:
Checking again, it does look like all configurations of pkparse call at least `mbedtls_oid_get_pk_alg`, and same with pkwrite 
calling `mbedtls_oid_get_oid_by_pk_alg`. So I must have been mistaken in #8167, and it's a simple `MBEDTLS_PK_{PARSE,WRITE}_C` requires `MBEDTLS_OID_C`.